lib/Segments/WooCommerce.php

```diff
@@ -80,8 +80,14 @@ public function synchronizeGuestCustomer($orderId, $currentFilter = null) {
     $wcSegment = Segment::getWooCommerceSegment();
 
     if ($wcOrder === false or $wcSegment === false) return;
+    $signupConfirmation = $this->settings->get('signup_confirmation');
+    $status = Subscriber::STATUS_UNCONFIRMED;
+    if ((bool)$signupConfirmation['enabled'] === false) {
```

Comment:
The specs don't mention that we should only use `unconfirmed` status if the confirmation email setting in on. My understanding is to always use `unconfirmed` status.

Reply:
@flakas I think you should decide this one because I am not sure.

When the sign-up confirmation is disabled, what should happen? Do we still keep the status `unconfirmed`?

Reply:
@amine-mp @pavel-mailpoet Good catch, I have not taken into account that "Signup confirmation" can be turned off.
As much as I would like to always force double opt-in, since we do permit the option to disable "Signup confirmation", for consistency's sake I believe we should follow the "Signup confirmation" setting.

So I think the behavior that Pavel implemented is good:
a) If "Signup confirmation" is enabled, we set the subscriber as unconfirmed, and in most cases we send them a confirmation email (in a different code part).
b) If "Signup confirmation" is disabled, we sync the customer as "subscribed" if the checkbox is checked.